ui/js/controllers/perf/compare.js

```diff
@@ -359,25 +368,34 @@ perf.controller('CompareResultsCtrl', [
                                                        { push_id: [$scope.newResultSet.id] });
                     }).then((resultMaps) => {
                         $scope.dataLoading = false;
-                        displayResults(originalResultsMap, resultMaps[$scope.newResultSet.id]);
+                        const newResult = resultMaps[$scope.newResultSet.id];
+                        if (newResult) {
+                          displayResults(originalResultsMap, newResult);
+                        }
+                        $scope.$apply();
                     });
                 });
             }
         }
         // TODO: duplicated in comparesubtestctrl
         function verifyRevision(project, revision, rsid) {
-            return ThResultSetModel.getResultSetsFromRevision(project.name, revision).then(
-                function (resultSets) {
-                    const resultSet = resultSets[0];
-                    // TODO: this is a bit hacky to pass in 'original' as a text string
-                    if (rsid === 'original') {
-                        $scope.originalResultSet = resultSet;
+
+            return PushModel.getList({ repo: project.name, revision })
+                .then(async (resp) => {
+                    if (resp.ok) {
+                        const { results } = await resp.json();
+                        const resultSet = results[0];
+                        // TODO: this is a bit hacky to pass in 'original' as a text string
+                        if (rsid === 'original') {
+                            $scope.originalResultSet = resultSet;
+                        } else {
+                            $scope.newResultSet = resultSet;
+                        }
                     } else {
-                        $scope.newResultSet = resultSet;
+                      $scope.errors.push(resp.statusText);
```

Comment:
Same question here about whether json response for the error would be more useful.